modules/Bio/EnsEMBL/Compara/PipeConfig/Parts/DiamondAgainstRef.pm

```diff
@@ -0,0 +1,105 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::PipeConfig::Parts::DiamondAgainstRef
+
+=head1 DESCRIPTION
+
+    This is a partial PipeConfig to Diamond search a member_id list against given blast_db
+
+=cut
+
+package Bio::EnsEMBL::Compara::PipeConfig::Parts::DiamondAgainstRef;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Hive::Version 2.4;
+use Bio::EnsEMBL::Hive::PipeConfig::HiveGeneric_conf; # For WHEN and INPUT_PLUS
+
+sub pipeline_analyses_diamond_against_refdb {
+    my ($self) = @_;
+
+    my %blastp_parameters = (
+        'diamond_exe' => $self->o('diamond_exe'),
+        'blast_params'  => "#expr(#all_blast_params#->[2])expr#",
+        'evalue_limit'  => "#expr(#all_blast_params#->[3])expr#",
+        'blast_db'      => $self->o('blast_db'),
+    );
+
+    return [
+        {   -logic_name         => 'diamond_blastp',
+            -module             => 'Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondBlastp',
+            -parameters         => {
+                %blastp_parameters,
+            },
+            -rc_name            => '500Mb_4c_20min_job',
+            -flow_into          => {
+               -1 => [ 'diamond_blastp_himem' ],  # MEMLIMIT
+               -2 => 'break_batch',
```

Comment:
Is breaking the batch after 20 min really useful ? How long would it take otherwise ?

Reply:
The average length of each job is ~2 minutes for the batches tested, I see the break_batch here as more of a fail-safe, because I don't think it would take anywhere near 20 minutes for any batches (none so far), so if it does, something has gone wrong, and it can be passed on so that individual members can be processed. I considered this because of my testing with the HMM BLASTps and some of the strange behaviours some jobs exhibited. It is possible that the break_batch and beyond could be removed with Diamond in the future.

Reply:
OK, so leave it for now :+1: